components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java

```diff
@@ -134,7 +136,8 @@ public BuildValidationData fetchBuildValidationData(String buildScanId) {
                     buildOutcomeFrom(attributes),
                     remoteBuildCacheUrlFrom(buildCachePerformance),
                     summarizeTaskExecutions(buildCachePerformance),
-                    toDuration(buildCachePerformance.getEffectiveProjectExecutionTime())
+                    toDuration(buildCachePerformance.getEffectiveProjectExecutionTime()),
+                    BigDecimal.valueOf(buildCachePerformance.getSerializationFactor())
```

Comment:
0: I'm curious about the decision to use BigDecimal here. Can a serialization factor get so big, or have so many precision points that it really needs to be represented as a BigDecimal?

Later on, we round this value to two decimal places, which suggests highly accurate precision is not required in this case.

Reply:
I usually just default to using `BigDecimal` when dealing with floating point values in Java.

Do you have preference for what to use instead?

Reply:
I think using BigDecimal is fine in this case. There some useful methods on BigDecimal that we're taking advantage of. I was mostly curious as to what the original reason was.